kythe/rust/indexer/tests/providers.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 The Kythe Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate kythe_rust_indexer;
+use kythe_rust_indexer::{
+  error::KytheError,
+  providers::{FileProvider, KzipFileProvider},
+};
+use std::env;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+/// Get the path of the provided runfile
+///
+/// # Panics
+///
+/// Will panic if the TEST_SRCDIR environment variable isn't set.
+fn get_runfile(file: &str) -> PathBuf {
+  let testfiles_path = "io_kythe/kythe/rust/indexer/testfiles/";
+  match env::var("TEST_SRCDIR") {
```

Comment:
Could this be based on `env!("CARGO_MANIFEST_DIR")` ([docs](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates)) instead for simplicity?

Reply:
Kythe is built using Bazel, so the indexer is built using the Rust rules rather than using Cargo. `TEST_SRCDIR` is set by Bazel when running the test target, which points to the location of the runfiles.

Reply:
I don't believe the Bazel rules set that variable.

Reply:
Ah, right. Might be a nice extension for bazel's rust support, but in the meantime this makes sense.

Reply:
It's set when you use the data parameter. https://docs.bazel.build/versions/master/test-encyclopedia.html

Reply:
rules_rust provides a runfiles accessor https://github.com/bazelbuild/rules_rust/blob/master/tools/runfiles/runfiles.rs#L25-L28